src/ivoc/ocmatrix.cpp

```diff
@@ -34,23 +34,35 @@ OcMatrix* OcMatrix::instance(int nrow, int ncol, int type) {
     }
 }
 
-void OcMatrix::unimp() {
-    hoc_execerror("Matrix method not implemented for this type matrix", 0);
+void OcMatrix::unimp() const {
+    hoc_execerror("Matrix method not implemented for this type matrix", nullptr);
 }
 
-void OcMatrix::nonzeros(std::vector<int>& m, std::vector<int>& n) {
+void OcMatrix::nonzeros(std::vector<int>& m, std::vector<int>& n) const {
     m.clear();
     n.clear();
     for (int i = 0; i < nrow(); i++) {
         for (int j = 0; j < ncol(); j++) {
-            if (getval(i, j) != 0) {
+            if (getval(i, j) != 0.) {
                 m.push_back(i);
                 n.push_back(j);
             }
         }
     }
 }
 
+std::vector<std::pair<int, int>> OcMatrix::nonzeros() const {
```

Comment:
it's slightly unfortunate that we need two different APIs for getting non-zeros, one using two separate vector and the other using a vector of pairs.

if it's easy to avoid, consider using just one version and then implementing one in terms of the other.

Reply:
I remove the old signature.